### PR TITLE
[enhancement] enable autocomplete for school and major fields

### PR DESCRIPTION
**Pull request by WilhelmWillie**
Screen Shot 2019-11-03 at 3 30 50 PM

There's a few different reasons why I want to implement autocomplete.

1. Better user experience, makes it easier to type out information
2. School + major will have high cardinality when sometimes it probably shouldn't. For example, "Computer Science" might show up as "CS", "computer science", "Computer science", etc. While this won't guarantee reduced cardinality, it'll likely mean that answers will be more clustered together which will make data analysis easier on our (organizers) end

Also, found some data sets from HackMIT Quill that we could use. Based off a user's e-mail, we can strongly infer the school they go to, so this includes a file that maps e-mail domains to schools (ex: usc.edu -> University of Southern California)

### CODE REVIEW

File: assets/data/majors.json [hunk 1]
```diff
@@ -0,0 +1,176 @@
+[
```
Comment (WilhelmWillie) [inline assets/data/majors.json]: this won't capture every major out there, but captures most general major categories. users can still enter in their own unique major if they have something that doesn't fit any of these

File: lib/useDebounce.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react";
```
Comment (WilhelmWillie) [inline lib/useDebounce.tsx]: source: usehooks.com

File: tsconfig.json [hunk 3]
```diff
@@ -21,5 +21,12 @@
...
     "**/*.test.tsx"
   ],
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "**/*.png", "**/*.svg"]
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    "**/*.png",
+    "**/*.svg",
+    "**/*.json"
```
Comment (WilhelmWillie) [inline tsconfig.json]: needed to import json or else typescript gets mad

File: assets/data/email-to-school.json [hunk 4]
```diff
@@ -0,0 +1,212 @@
+{
```
Comment (saresend) [inline assets/data/email-to-school.json]: Hot diggity dog, nice

**Pull request merged**